### DISCUSSION TITLE
Error compiling ANURA3D_2024

### DISCUSSION THREAD

**Discussion by 19Dario88** [Anura3D/Anura3D_OpenSource#235] [Q&A] [answered]
Dear all,
I'm trying to compile the last version of ANURA3D (2024), as it was released few days ago (first of all: thank you for your efforts during the workshop!). I installed Visual Studio 2022 (v.17.9.6), OneAPI Base+HPC toolkit (v. 2024.2), and everything worked fine when compiling the previous version of the code (2023).
Now, thousands of warnings and 17 errors appear, and compilation is aborted.
Could you suggest how to solve the problem? Could it depend by the software's (VS+API) versions?

[BuildLog.txt](https://github.com/user-attachments/files/16069517/BuildLog.txt)
![Error compiling A3D_2024](https://github.com/Anura3D/Anura3D_OpenSource/assets/168671577/2c86f8ed-289d-4e36-a2ef-095a654e37f9)
Thank you in advance for any advice!
Regards, 
Dario

**Comment by marcusdelfino** [comment]
Hello, Dario.

I'm having the same problem as you.
Looking forward to the answer.

Marcus

**Reply by pmarveggio** [reply]
Hi Marcus,
which version are you using?
@marcusdelfino and @19Dario88 can you check whether you have the WriteOutPut_GiD.f file in the src folder?
Pietro

**Reply by 19Dario88** [reply]
Hi @pmarveggio,
Yes, I have that file in the folder. I am worried it could depend on the version of the Intel's toolkits (2024), because the user @emre-ncche (#233 ) is able to compile the last release of code with a previous version of them. However, older version of the toolkits cannot be downloaded no more from the Intel's website.
Thank you in advance for your interest.
Regards,
Dario

**Reply by sabrina14chen** [reply]
@pmarveggio 我的src資料夾中確定有WriteOutPut_GiD.f檔案，但還是在用的時候出現錯誤

**Comment by Mangwuha** [comment]
Hello everyone, 
I have  similar situations in the process of installation and compilation.
1. First of all, when I installed oneAPI2024.2 on my desktop computer, I kept getting errors, as shown in the first picture below. This is true for both 2024.2 and 2024.1. That didn't happen on my laptop, both of which run windows.
2. Second, when I compile "Anura3D.sln" on my laptop using VS2022, I get an error saying that "Fortran compiler (.exe) "cannot be found.
I wonder if anyone can help me. thank you.
![EK}@ 4Y@H2ESQ X1D_ 7L7](https://github.com/Anura3D/Anura3D_OpenSource/assets/139300915/c2dd1ae0-5063-4e5e-b0ec-b2970e8c3538)

**Reply by 19Dario88** [reply]
Hi Mangwuha,
regarding the second point, have you also installed the oneAPI HPC Toolkit, before trying to compile the code?
If this issue is solved, can you please give us notice if you can successfully compile the code with the 2024 version of the OneAPI toolkits (Base+HPC)?
Regards,
Dario

**Reply by Mangwuha** [reply]
Hello Dario,
I have installed the oneAPI HPC Toolkit, and the result is as same as the base toolkit, just like the picture above, some errors there.
I don't know what caused the failure to install oneAPI properly. So I didn't compile it successfully.I'm also asking for help. As soon as I can successfully resolve the problem, I will contact you.
Best wishes.

**Reply by 19Dario88** [reply]
Hi Mangwuha,
maybe you can share the log file, some expert users could help you.
Regards,
Dario

**Reply by pmarveggio** [reply]
@Mangwuha can you share the log file?
Thanks!

**Reply by HM-3214** [reply]
@Mangwuha  I also met similar warning when installing Intel Toolkit using online installation. When I changed to offline installaition, this problem was solved. Regarding your second point, I think it was because you did not sucessfully integrate Fortran into VS yet, due to the first issue you met.

**Reply by weilinlee0308** [reply]
@pmarveggio Hi, thanks for your kind help. I have another question. The file "TutorialManual_2024.pdf" cannot be downloaded using the following link: https://github.com/Anura3D/Anura3D_OpenSource/blob/main/manuals/TutorialManual_2024.pdf Thanks

**Reply by pmarveggio** [reply, marked as answer]
Hi everyone, thanks for all the reply. The issue is indeed related to the oneAPI2024.2.
To fix it, you should comment line 55 of WriteOutput_Gid.for file by adding a ! before the command:

>     !#pragma comment(lib, "gidpost.lib")

This will disable the option to post-process results in GiD. We are working on that and will release a patch soon.
Please let me know if this solution works on your system too.
Thanks all for the support.

**Reply by FelipeMedan** [reply]
> Hi everyone, thanks for all the reply. The issue is indeed related to the oneAPI2024.2. To fix it, you should comment line 55 of WriteOutput_Gid.for file by adding a ! before the command:
> 
> > ```
> > !#pragma comment(lib, "gidpost.lib")
> > ```
> 
> This will disable the option to post-process results in GiD. We are working on that and will release a patch soon. Please let me know if this solution works on your system too. Thanks all for the support.

This worked for me, thanks! Looking forward to that patch!

**Reply by weilinlee0308** [reply]
This worked for me, too. Thanks!!!

**Reply by 19Dario88** [reply]
@pmarveggio,it works! thank you

**Reply by Mangwuha** [reply]
@pmarveggio ,
Hi,I tried the solution you proposed, but my problem still hasn't been solved, so I uploaded my error report file. It's the same problem I had before.
![0O)6(%%XIHBS7(%_I}0~29J](https://github.com/user-attachments/assets/29ef64c1-670b-45b0-b34e-ad210208e1d0)

**Reply by Mangwuha** [reply]
@pmarveggio ,
On my computer, VS2022 and oneAPI have been successfully integrated.
![微信截图_20240722121652](https://github.com/user-attachments/assets/4ac87af8-53fe-4bbb-a8ac-882710d3a7b5)

**Reply by jiawei6688** [reply]
> Hi everyone, thanks for all the reply. The issue is indeed related to the oneAPI2024.2. To fix it, you should comment line 55 of WriteOutput_Gid.for file by adding a ! before the command:
> 
> > ```
> > !#pragma comment(lib, "gidpost.lib")
> > ```
> 
> This will disable the option to post-process results in GiD. We are working on that and will release a patch soon. Please let me know if this solution works on your system too. Thanks all for the support.

Excuse me, where can I find the file to add the code? I don't have what you said.

**Comment by coarse-snow** [comment]
@19Dario88 hello, I installed the same version of Visual Studio 2022 (v.17.9.6),  OneAPI Base+HPC toolkit (v. 2024.2) as you, and I met same problem when compiling the Anura2024, but the same problem also occurred in Anura2023, can you give some suggestions, thanks
![image](https://github.com/user-attachments/assets/2d482e1c-e558-4e6b-acd2-8846d55104fc)

**Comment by FelipeMedan** [comment]
Hello Dario,

I'm having the same problem and I'm using the same versions of VS and oneAPI Base + HPC. If someone has any suggestions, I would really appreciate it.

Thanks!

![Screenshot 2024-07-16 152318](https://github.com/user-attachments/assets/7d93afbf-0218-4d46-9a27-12cd4bf7839e)

**Comment by FJPVUCHILE** [comment]
Hello,

I am experiencing the same problem mentioned above. Has anyone found a solution?

Your assistance would be greatly appreciated.

Thank you!

**Reply by weilinlee0308** [reply]
I compiled the source code with VS2017 and OpenAPI 2022 on Windows 10 successfully.

https://www.dropbox.com/scl/fi/u9ot1rolpzhhgebhyr7y4/Anura3D_2024.gid.zip?rlkey=1445x3hvyxia47x1qihz7mvha&st=mew265d5&dl=0

**Comment by marcusdelfino** [comment]
Does anyone have problem with the release file now?

The executable file right now doesn't work because the file "zlib.dll" is not installed.

**Reply by FelipeMedan** [reply]
> I compiled the source code with VS2017 and OpenAPI 2022 on Windows 10 successfully.
> 
> https://www.dropbox.com/scl/fi/u9ot1rolpzhhgebhyr7y4/Anura3D_2024.gid.zip?rlkey=1445x3hvyxia47x1qihz7mvha&st=mew265d5&dl=0

Hi Marcus,

I get the same issue. I was able to compile and get the .exe file and paste it in the exec folder, but when I try to run the calculate file I get that error. What I'm doing now is just using the compiled files that @weilinlee0308 uploaded in this same thread. That worked for me.

**Reply by sabrina14chen** [reply]
我也有同樣的問題

**Reply by GBarbosaS** [reply]
First, I compiled the code with Visual Studio 2022 and OpenAPI 2024.2 on Windows 11. The code compiled, but the executable does not open, showing the error: (0xc000007b). The calculate.bat files generate the error: not recognized as an internal or external command, operable program or batch file. I tried using the compiled file provided on Dropbox, but it still didn't work. Could you provide details on how you got yours to work with the file already available on Dropbox?

**Reply by weilinlee0308** [reply]
https://www.dropbox.com/scl/fi/o97drus1swvpzkgj95u24/Anura3D_2025.gid.zip?rlkey=o9ik5sj7imq3dh7uqx8p5ellr&dl=0

Here is a working project file of v2025. Please give it a try and carefully review the public manual and YouTube tutorials.

**Reply by Armin-Geranghadr** [reply]
> First, I compiled the code with Visual Studio 2022 and OpenAPI 2024.2 on Windows 11. The code compiled, but the executable does not open, showing the error: (0xc000007b). The calculate.bat files generate the error: not recognized as an internal or external command, operable program or batch file. I tried using the compiled file provided on Dropbox, but it still didn't work. Could you provide details on how you got yours to work with the file already available on Dropbox?



I got almost similar same error:

'C:\ANURA' is not recognized as an internal or external command,
operable program or batch file.
'C:\ANURA' is not recognized as an internal or external command,
operable program or batch file.
    while executing
"exec $batchfile $projectName [file nativename $projectfolder]  [file nativename $dir] [file nativename $A3D_folder] "
    (procedure "Anura3D::Calculate" line 276)
    invoked from within
"Anura3D::Calculate local"
    (menu invoke)
    
  
Could anyone help me how to address this issue?

**Reply by elbarrendero** [reply]
> https://www.dropbox.com/scl/fi/o97drus1swvpzkgj95u24/Anura3D_2025.gid.zip?rlkey=o9ik5sj7imq3dh7uqx8p5ellr&dl=0
> 
> Here is a working project file of v2025. Please give it a try and carefully review the public manual and YouTube tutorials.

i've compiled the exact same solution... glad that's okay... i've got some issues though running the ANURA3D_2025 tutorials, given that i'm able to generate the input files, but the vtk output files haven been able to generate them correctly, only one time and had just one step of data...

**Comment by EmreDumlu-UM** [comment]
Hi Armin,
The error you run into says there is a problem with the compiling Anura3D.
Please check the error messages in Visual Studio. It should say 1 succeed,
0:error, after compiling the release.
If you compiled with success, please be sure that you copy Anura3D
executable file in .gid file then you copy and paste it into problem
solution file in the main folder of GiD

**Reply by Armin-Geranghadr** [reply]
Thanks for your response. Once again, I compiled the code and 1 succeed, 0:error message. Also, I copied Anura3D
executable file, but the problem still remains. Do you have any other solutions?

Once again, I appreciate your contribution

**Reply by EmreDumlu-UM** [reply]
You should have copied the executable file (Anura3D_2024.exe) in the release folder into Anura3D_2024.gid folder which is located at the GiD_Problemtype, after src. Then you copied Anura3D_2024.gid problem types folder in GiD. Is that correct? If it's so, there is no problem in this part.
Apart from above, your GiD may not generate Anura3D folder. May be due to the version issue of GiD. Could you use GiD V16 to generate these files?

**Comment by Armin-Geranghadr** [comment]
Dear Emre,

I installed GiD 17.0.2 instead of GiD 17.1.2. Also, As you can see, I
copied the executable file into Anura3D_2024.gid and then copied
Anura3D_2024.gid problem types folder in GiD. Although, the error still
remains, and as you can see in the second picture, after I ran the model
just these files had been generated. I want to sketch my model from the
start point. Do you have any other suggestions?

Regards.
